### DISCUSSION TITLE
Query for all active alerts for a specified coverage area (issued by a specified WFO)?

### DISCUSSION THREAD

**Discussion by ckolovson** [weather-gov/api#679] [Q&A]
I'm trying to create an API call to get all active alerts issued by our local WFO (MTR). Ideally, it would be as simple as:
```
https://api.weather.gov/alerts/active?cwa=mtr
```
but currently there is no parameter for coverage area. What I'm doing instead is specifying all the counties in our CWA:
```
https://api.weather.gov/alerts/active?zone=CAC081,CAC085,CAC001,CAC013,CAC041,CAC053,CAC055,CAC069,CAC075,CAC087,CAC095,CAC097
```
Is this the best way of doing it, and will this result in duplicates?

**Comment by ckolovson** [comment]
I think I have a solution:
```
import requests
from datetime import datetime

URL = 'https://api.weather.gov/alerts/active?area=CA'
response = requests.get(URL)
alerts = response.json()['features']
nactive = 0

print(f'There are {len(alerts)} active alerts in CA.')

for alert in alerts:
    endsTime = datetime.fromisoformat(alert['properties']['ends'])
    senderName = alert['properties']['senderName']
    if senderName.lower() == "NWS San Francisco CA".lower():
      print(f"{alert['properties']['event']} for {alert['properties']['areaDesc']} till {endsTime.strftime('%A, %B %d, %Y at %I:%M %p %Z')}")
      print()
      nactive += 1

if nactive > 0:
  print(f'There are', nactive, 'active alerts issued by MTR.')
else: 
  print(f'There are currently no active alerts issued by MTR.')
```
At the time I called it, the output of this script was:
```
There are 7 active alerts in CA.

Dense Fog Advisory for Northern Salinas Valley/Hollister Valley and Carmel Valley; Northern Monterey Bay; Southern Monterey Bay and Big Sur Coast till Monday, December 04, 2023 at 01:00 AM UTC-08:00

Beach Hazards Statement for San Francisco; Coastal North Bay Including Point Reyes National Seashore; San Fransisco Peninsula Coast; Southern Monterey Bay and Big Sur Coast till Monday, December 04, 2023 at 01:00 AM UTC-08:00

There are 2 active alerts issued by MTR.
```
I'm not sure if this will or won't return duplicates. I'm still waiting for an answer to https://github.com/weather-gov/api/discussions/678. Although this works, it would be preferable if the API could support such a query directly? Is this possible, or could this be added?

**Reply by dwhitemv25** [reply]
A few observations: 

1. The list of zones covered by a WFO is listed on the `/offices/<WFO>` query. Be aware that WFO coverage areas are not limited to a single state, and alerts can be issued for any of the 3 zone categories (forecast, fire, county). Marine zones are their own issue.
2. My brief experiments have not generated any duplicate alerts in a single query response.
3. The WFO ID is embedded in the `AWIPSIdentifier` parameter, which might be easier to scan for, at least for alerts that are sent with one. Alerts originated from outside NWS and relayed by the WFO may not have one (Air Quality Alerts, for instance).

**Reply by ckolovson** [reply]
Thank you for that info. It would be great if you guys could add a new filter parameter to the /alerts/active API based on a WFO's coverage area (CWA). For example, there could be: /alerts/active?cwa=MTR. Please consider this as an enhancement request. This would be similar to the URL below (used on the https://www.weather.gov/mtr/ web page):
```
https://www.weather.gov/wwamap/wwatxtget.php?cwa=mtr&wwa=all
```
Note however that the above URL returns alerts for all of California, so unfortunately it doesn't work as one would expect. I realize that this (legacy?) URL is not part of the NWS API Web Service. I was just citing its "cwa=" filter parameter as an example.

**Comment by ckolovson** [comment]
Specifying a list of counties works, as in:
```
URL = 'https://api.weather.gov/alerts/active?zone=CAC001,CAC013,CAC041,CAC053,CAC055,CAC069,CAC075,CAC081,CAC085,CAC087,CAC097'
```
